### PR TITLE
chore!: upgrade PSS to SDK v0.50.x  (ICS v4.3.x features only)

### PR DESCRIPTION
**Pull request by sainoe**
This PR upgrades the PSS features present in `ICS v4.3.x` to `SDK v0.50.x`, by introducing PSS into `ICS release/5.x` 
(the target [release/v5.1.x](https://github.com/cosmos/interchain-security/tree/release/v5.1.x) branch was created from [release/v5.x](https://github.com/cosmos/interchain-security/tree/release/v5.x)).

Closes: [#3120](https://github.com/cosmos/gaia/issues/3120)

Credit to @bermuell for his big contribution to this upgrade, especially for backporting all the E2E tests. 
## Description

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

* [ ] Included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
* [ ] Added `!` to the type prefix if the change is [state-machine breaking](https://github.com/cosmos/interchain-security/blob/main/RELEASES.md#breaking-changes)
* [ ] Confirmed this PR does not introduce changes requiring state migrations, OR migration code has been added to consumer and/or provider modules
* [ ] Targeted the correct branch (see [PR Targeting](https://github.com/cosmos/interchain-security/blob/main/CONTRIBUTING.md#pr-targeting))
* [ ] Provided a link to the relevant issue or specification
* [ ] Followed the guidelines for [building SDK modules](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/00-intro.md)
* [ ] Included the necessary unit and integration [tests](https://github.com/cosmos/interchain-security/blob/main/CONTRIBUTING.md#testing)
* [ ] Added a changelog entry to `CHANGELOG.md`
* [ ] Included comments for [documenting Go code](https://blog.golang.org/godoc)
* [ ] Updated the relevant documentation or specification
* [ ] Reviewed "Files changed" and left comments if necessary 
* [ ] Confirmed all CI checks have passed
* [ ] If this PR is library API breaking, bump the go.mod version string of the repo, and follow through on a new major release

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

* [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
* [ ] confirmed `!` the type prefix if the change is state-machine breaking
* [ ] confirmed this PR does not introduce changes requiring state migrations, OR confirmed migration code has been added to consumer and/or provider modules
* [ ] confirmed all author checklist items have been addressed
* [ ] reviewed state machine logic
* [ ] reviewed API design and naming
* [ ] reviewed documentation is accurate
* [ ] reviewed tests and test coverage

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,8 @@
 
 ## v5.0.0
 
+❗ *The provider module should never not be used with this version.*
```
Comment (p-offtermatt) [inline CHANGELOG.md:5]:
```suggestion
❗ *The provider module should not be used with this version.*
```
Double negation! :D I guess this has to be changed in summary.md ultimately

File: proto/interchain_security/ccv/provider/v1/tx.proto [hunk 2]
```diff
@@ -101,7 +103,7 @@ message MsgUpdateParamsResponse {}
 
 
 // MsgConsumerAddition defines the message used to spawn a new consumer chain using a v1 governance proposal.
-// If it passes, then all validators on the provider chain are expected to validate
+// If it passes, then all ˝validators on the provider chain are expected to validate
```
Comment (p-offtermatt) [inline proto/interchain_security/ccv/provider/v1/tx.proto:106]:
```suggestion
// If it passes, then all validators on the provider chain are expected to validate
```

**Review by p-offtermatt — COMMENTED** [review]
Hey, shouldn't GetLastValidators be removed from the expected keeper here?
https://github.com/cosmos/interchain-security/blob/2f01f9c8e67bceb2c231d3e62351d3d20eeaec3e/x/ccv/types/expected_keepers.go#L50

**Review by p-offtermatt — APPROVED** [review]
Just one comment. Otherwise, I did not review too closely (since everything going into this was already reviewed), but surface-level everything seems in order

File: proto/interchain_security/ccv/provider/v1/tx.proto [hunk 3]
```diff
@@ -210,3 +229,97 @@ message MsgChangeRewardDenoms {
...
+  string chain_id = 1;
+  // the validator address on the provider
+  string provider_addr = 2 [ (gogoproto.moretags) = "yaml:\"address\"" ];
+  // (optional) The consensus public key to use on the consumer in json string format corresponding to proto-any,
+  // for example `{"@type":"/cosmos.crypto.ed25519.PubKey","key":"Ui5Gf1+mtWUdH8u3xlmzdKID+F3PK0sfXZ73GZ6q6is="}`.
+  // This field is optional and can remain empty (i.e., `consumer_key = ""`). A validator can always change the
+  // consumer public key at a later stage by issuing a `MsgAssignConsumerKey` message.
+  string consumer_key = 3;
+  // signer address
+  string signer = 4 [(cosmos_proto.scalar) = "cosmos.AddressString"];
```
Comment (insumity) [inline proto/interchain_security/ccv/provider/v1/tx.proto:247]: I guess `signer` here is due to SKD v0.50?
Why is it called `authority` in some messages but `signer` here?
Also, shouldn't `MsgSetConsumerCommissionRate` have a `signer`?

File: x/ccv/provider/keeper/distribution.go [hunk 4]
```diff
@@ -57,32 +66,273 @@ func (k Keeper) GetAllConsumerRewardDenoms(ctx sdk.Context) (consumerRewardDenom
...
+
+		// note that it's possible that no rewards are collected even though the
+		// reward pool isn't empty. This can happen if the reward pool holds some tokens
+		// of non-whitelisted denominations.
+		alloc := k.GetConsumerRewardsAllocation(ctx, consumerChainID)
+		if alloc.Rewards.IsZero() {
+			continue
+		}
+
+		// if rewardsCollected.IsZero() {
```
Comment (insumity) [inline x/ccv/provider/keeper/distribution.go:88]: Why is this commented out?

File: x/ccv/provider/keeper/consumer_equivocation.go [hunk 5]
```diff
@@ -367,23 +367,27 @@ func (k Keeper) JailAndTombstoneValidator(ctx sdk.Context, providerAddr types.Pr
...
+		err := k.stakingKeeper.Jail(ctx, providerAddr.ToSdkConsAddr())
+		if err != nil {
+			return err
+		}
 	}
 
-	k.slashingKeeper.JailUntil(ctx, providerAddr.ToSdkConsAddr(), evidencetypes.DoubleSignJailEndTime)
+	err = k.slashingKeeper.JailUntil(ctx, providerAddr.ToSdkConsAddr(), evidencetypes.DoubleSignJailEndTime)
+	if err != nil {
+		return fmt.Errorf("fail to set jail duration for validator: %s", providerAddr.String())
```
Comment (insumity) [inline x/ccv/provider/keeper/consumer_equivocation.go:383]: Should we wrap `err` in this case?

File: x/ccv/provider/keeper/distribution.go [hunk 6]
```diff
@@ -57,32 +66,273 @@ func (k Keeper) GetAllConsumerRewardDenoms(ctx sdk.Context) (consumerRewardDenom
...
+		consAddr := sdk.ConsAddress(consumerVal.ProviderConsAddr)
+
+		// get the validator tokens fraction using its voting power
+		powerFraction := math.LegacyNewDec(consumerVal.Power).QuoTruncate(totalPower)
+		tokensFraction := tokens.MulDecTruncate(powerFraction)
+
+		// get the validator type struct for the consensus address
+		val, err := k.stakingKeeper.GetValidatorByConsAddr(ctx, consAddr)
+		if err != nil {
+			k.Logger(ctx).Error("cannot find validator by consensus address :%s while allocating rewards from consumer chain: %s",
```
Comment (insumity) [inline x/ccv/provider/keeper/distribution.go:213]: Here and below: Does string formatting (e.g., `%s`) work together with `k.Logger(ctx)`?

File: x/ccv/provider/keeper/distribution.go (hunk 4 above)
Comment (sainoe) [reply x/ccv/provider/keeper/distribution.go:88]: God catch! It's not needed anymore.

File: x/ccv/provider/keeper/distribution.go (hunk 6 above)
Comment (sainoe) [reply x/ccv/provider/keeper/distribution.go:213]: It doesn't, good catch.

File: proto/interchain_security/ccv/provider/v1/tx.proto (hunk 3 above)
Comment (sainoe) [reply proto/interchain_security/ccv/provider/v1/tx.proto:247]: IIUC `signer` is for normal msg and `authority` for proposals.

**Review by insumity — APPROVED** [review]

**Pull request merged**